test/Streamly/Test/Prelude/Serial.hs

```diff
@@ -674,3 +710,5 @@ main = hspec
 
     describe "Nesting" $ do
         prop "foldIterateM" foldIterateM
+
+    describe "Tests for retry" retry
```

Comment:
Shouldn't  this go under `"Tests for exceptions"` above? Also the code block should be moved next to `exceptionOps`.